graphql_compiler/schema_generation/sqlalchemy/utils.py

```diff
@@ -1,18 +1,32 @@
 # Copyright 2019-present Kensho Technologies, LLC.
+from typing import Iterable, List, Set
+
+from sqlalchemy import Table
+
 from ..exceptions import MissingPrimaryKeyError
 
 
-def validate_that_tables_have_primary_keys(tables):
+def validate_that_tables_have_primary_keys(tables: Iterable[Table]):
     """Validate that each SQLAlchemy Table object has a primary key."""
+    tables_missing_primary_keys: Set[Table] = set()
```

Comment:
It doesn't hurt and now that you added you don't have to remove them, but you don't need to add type annotations to all variables within a function. I think that our convention is to add type annotations in:
1. Function signatures. 
2. Global variables

Mypy might also sometimes require additional type annotations to function properly, (you'll know when this is the case because mypy will raise some sort of error or warning).

Reply:
Suggestion instead of making this a set of tables, make this a set of strings.

Then write `table_missing_primary_keys.add(table.fullname)` below. The fullname includes both the name and the schema information. 

Then at the bottom you can write simply write:
`raising MiisingPrimaryKeyError(
... 
f"Tables missing primary keys: {tables_missing_primary_keys}"
)`

Notice that I used an f-string. This is a new python feature that we are starting to use as a replacement for `format()`. 
https://realpython.com/python-f-strings/

I won't block your review if you decide to use `format()` instead, but f-strings are more convenient to use so I think it'd be a good idea if you got acquainted with f-strings.

Reply:
+1 to @pmantica1's points. This is fine, working code, but it is slightly unidiomatic: we're targeting Python 3.6+ and not using the best available features in the target language. Let's go for one more round of polish since this is code that our users are likely to read — it raises a non-obvious user-facing error, for which it's not unreasonable to dig into the source.

Reply:
It turns out that this particular annotation would be good (to clarify that it's a set of strings in particular rather than a set of some unknown type), but that's a good point in general-- thanks for the heads-up!

The f-string thing is really handy, I've updated it.